workers/src/executor_def.rs

```diff
@@ -0,0 +1,152 @@
+use crate::abi::L0V2EndpointAbi::PacketSent;
+use crate::abi::L0V2EndpointAbi::PacketVerified;
+use crate::abi::SendLibraryAbi::ExecutorFeePaid;
+use crate::chain::connections::build_executor_subscriptions;
+use crate::chain::connections::get_abi_from_path;
+use crate::chain::connections::get_http_provider;
+use crate::chain::contracts::create_contract_instance;
+use crate::chain::contracts::{lz_receive, prepare_header};
+use crate::chain::ContractInst;
+use crate::config::DVNConfig;
+use alloy::dyn_abi::DynSolValue;
+use alloy::primitives::I256;
+use eyre::Result;
+use futures::StreamExt;
+use std::collections::VecDeque;
+use std::time::Duration;
+use tokio::time::sleep;
+use tracing::error;
+
+pub struct NFFLExecutor {
+    config: DVNConfig,
+    packet_queue: VecDeque<PacketSent>,
+    finish: bool,
+}
+
+impl NFFLExecutor {
```

Comment:
There are not tests on this PR. A bit scary tbh, do we plan to have some tests at some point?

Reply:
I really tried to do it with mocks, but I'd better do an isolated integration test with some live data.

Reply:
@coderabbit can you make an issue regarding executor integration tests?